### ISSUE TITLE
No visable assets in fledge when using report control block

### ISSUE THREAD

**Issue by JakobVogelsang** [fledge-power/fledge-south-iec61850#37]
I am currently working with the SCL file below and am I simulating using IEDScout 5.20. Below the configuration of the plugin.
I do the communication in the sniffer also following the trigger options in the report. No of it is visible in the assets. When I take the same server and remove in the config anything related to the report and simply poll the data I so see assets being added.

**Config**
```json
{
 "protocol_stack": {
  "name": "iec61850client",
  "version": "0.0.1",
  "transport_layer": {
   "ied_name": "IED1",
   "connections": [
    {
     "ip_addr": "192.168.0.67",
     "port": 49152,
     "tls": false
    }
   ]
  },
  "application_layer": {
   "polling_interval": 5000,
   "datasets": [
    {
     "dataset_ref": "IEDLDevice/LLN0.SwitchGear",
     "entries": [
      "IEDLDevice/XCBR1.Pos.stVal[ST]",
      "IEDLDevice/XCBR1.Pos.q[ST]"
     ]
    }
   ],
   "report_subscriptions": [
    {
     "rcb_ref": "IEDLDevice/LLN0.BR.SwitchGear",
     "dataset_ref": "IEDLDevice/LLN0.SwitchGear"
    }
   ]
  }
 }
}
```

```json
{
 "exchanged_data": {
  "datapoints": [
   {
    "pivot_id": "TS1",
    "label": "TS1",
    "protocols": [
     {
      "name": "iec61850",
      "objref": "IEDLDevice/XCBR1.Pos",
      "cdc": "DpsTyp"
     }
    ]
   }
  ]
 }
}
```

**PCAP**:

[WorkingReportControl.pcap.zip](https://github.com/fledge-power/fledge-south-iec61850/files/14239066/WorkingReportControl.pcap.zip)

**Server definition**
[TestIED (2).scd.zip](https://github.com/fledge-power/fledge-south-iec61850/files/14239027/TestIED.2.scd.zip)

**LOG**:
<img width="1561" alt="grafik" src="https://github.com/fledge-power/fledge-south-iec61850/assets/66802940/acbfe0fa-63bd-44d7-893b-84b9395a6ef5">

**Comment by mzillgith** [comment]
I cannot reproduce it. Maybe it has been fixed in the meantime. Can you try again with the latest code in the "develop_mz" branch?

**Comment by JakobVogelsang** [comment]
I did test that with `1.0.0-rc1`, sorry. Will retest and report back

**Comment by JakobVogelsang** [comment]
No, unfortunately, I still can reproduce this scenario. Report are visible in the bus, but there are no assets generated. To check the proper `exchanged_data` config, I just remove all `datasets` and `report_subscriptions` and assets are bing generated.

**Comment by JakobVogelsang** [comment]
Actually there seem to be a new error with `SetBRCBValues` such that the trigger options are not set correctly.

**Comment by jude-adam** [comment]
When testing with the provided SCD files and configurations, I also noticed that there was a SetRCBValues error. I think this was caused by the "rcb_ref" value being wrong, it works with IEDLDevice/LLN0.BR.SwitchGear01, not IEDLDevice/LLN0.BR.SwitchGear. Also, it was also noted that sending assets for the quality  attribute reports was not yet implemented. This has now been fixed and will be pushed soon to the develop_mz branch.

**Comment by JakobVogelsang** [comment]
Once you have it going I would love it when you share with me a configuration that I can test, either with a new SCL file or any other SCL file that you are using for testing.

**Comment by jude-adam** [comment]
The fix has been pushed to the develop_mz branch.

Configuration :
The SCL file I used was the one you provided.
Here is the config I used:

```json
{
 "protocol_stack": {
  "name": "iec61850client",
  "version": "0.0.1",
  "transport_layer": {
   "ied_name": "IED1",
   "connections": [
    {
     "ip_addr": "192.168.0.134",
     "port": 102,
     "tls": false
    }
   ]
  },
  "application_layer": {
   "polling_interval": 0,
   "datasets": [
    {
     "dataset_ref": "IEDLDevice/LLN0.SwitchGear",
     "entries": [
      "IEDLDevice/XCBR1.Pos.stVal[ST]",
      "IEDLDevice/XCBR1.Pos.q[ST]"
     ],
     "dynamic": false
    }
   ],
   "report_subscriptions": [
    {
     "rcb_ref": "IEDLDevice/LLN0.BR.SwitchGear01",
     "dataset_ref": "IEDLDevice/LLN0.SwitchGear"
    }
   ]
  }
 }
}
```

```json
{
 "exchanged_data": {
  "datapoints": [
   {
    "pivot_id": "TS1",
    "label": "TS1",
    "protocols": [
     {
      "name": "iec61850",
      "objref": "IEDLDevice/XCBR1.Pos",
      "cdc": "DpcTyp"
     }
    ]
   }
  ]
 }
}
```

Here is the Logger Output (Sends readings for stVal and q reports):

![image](https://github.com/fledge-power/fledge-south-iec61850/assets/145360131/5e146093-9919-479d-ac42-c7ac8b6ae82f)

**Comment by jude-adam** [comment]
@JakobVogelsang Any updates on testing so that the issue can be closed please.

**Comment by JakobVogelsang** [comment]
I am sorry but no I cannot confirm the report is being processed correctly. :( I see repots being activated and data is exchanged between the server and the client, but the information is not being processed correctly in the fledge instance. The assets are not showing any new incoming information.

Btw I am directly pulling from the branch  `develop_mz`. Is there a way how I can see a new version being used?

**Comment by JakobVogelsang** [comment]
What do I need to do to see the log message you have shown above?

**Comment by jude-adam** [comment]
Did you set the plugin's logger level to debug?

**Comment by JakobVogelsang** [comment]
Hi @jude-adam,

@mzillgith has had the winning hint. The plugin at the moment assumes that report references on the wire are the not indexed. Which means that it is expecting the above report control block reference to be as such:
`IEDLDevice/LLN0$BR$SwitchGear`

In my case however the simulation tool is indexing the report and the reference is changed to:
`IEDLDevice/LLN0$BR$SwitchGear01`

As a result, the references do not match and therefore the plugin does not receive reports.

I personally think that the report control block reference indexing is correct. How else would you be able to differentiate between `IEDLDevice/LLN0$BR$SwitchGear01` and `IEDLDevice/LLN0$BR$SwitchGear02`.

See my simple fix here https://github.com/JakobVogelsang/fledge-south-iec61850/commit/327410ab5e6e9ad090642b5f3424f66773e742a9

**Comment by Sander3003** [comment]
Great investigation!

**Comment by JakobVogelsang** [comment]
@mzillgith  has this been fixed now?

**Comment by mzillgith** [comment]
Hi @JakobVogelsang 
It doesn't seem to be fixed in the main branch. Your fix https://github.com/JakobVogelsang/fledge-south-iec61850/commit/327410ab5e6e9ad090642b5f3424f66773e742a9 seems to be correct.
I suggest you merge it into main.

**Comment by JakobVogelsang** [comment]
@mzillgith I did and would be happy about a review. BUT how did this work on your side, up until now. Do we not break something on the other side with this fix?

**Comment by mzillgith** [comment]
I don't think that it's breaking anything on the "other side" when you mean the server as the other side.

The only thing it could break as I see are the reporting related test cases. Need to check this.

**Comment by JakobVogelsang** [comment]
@mzillgith have you had time to check?

**Comment by mzillgith** [comment]
Sorry. I just checked now. I don't see any problem and the test cases including the reporting tests work without problems.
I merged your pull request so I think we can close this issue.

**Comment by Sander3003** [comment]
@JakobVogelsang can this be closed?
